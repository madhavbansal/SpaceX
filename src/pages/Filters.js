import React from 'react';
import { FilterSection } from './index';
import { sectionConfig } from '../utils/helper';
import { constants } from '../utils/constant';



const Filters = () => {
    return (
        <>
            {constants.filters}
            <FilterSection sectionConfig={sectionConfig.launchYear} />
            <FilterSection sectionConfig={sectionConfig.sucessfulLaunch} />
            <FilterSection sectionConfig={sectionConfig.sucessfulLanding} />

        </>
    )


}

export default React.memo(Filters);