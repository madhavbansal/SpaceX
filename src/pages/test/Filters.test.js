import React from 'react';
import renderer from 'react-test-renderer';
import Filters from '../Filters';


jest.mock('../FilterSection', ()=> 'FilterSection')

test('Test the Filter Component', () => {
  const component = renderer.create(
    <Filters/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});