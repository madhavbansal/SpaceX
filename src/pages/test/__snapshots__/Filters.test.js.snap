// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  "Filters",
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "2006",
            "label": "2006",
          },
          Object {
            "id": "2007",
            "label": "2007",
          },
          Object {
            "id": "2008",
            "label": "2008",
          },
          Object {
            "id": "2009",
            "label": "2009",
          },
          Object {
            "id": "2010",
            "label": "2010",
          },
          Object {
            "id": "2011",
            "label": "2011",
          },
          Object {
            "id": "2012",
            "label": "2012",
          },
          Object {
            "id": "2013",
            "label": "2013",
          },
          Object {
            "id": "2014",
            "label": "2014",
          },
          Object {
            "id": "2015",
            "label": "2015",
          },
          Object {
            "id": "2016",
            "label": "2016",
          },
          Object {
            "id": "2017",
            "label": "2017",
          },
          Object {
            "id": "2018",
            "label": "2018",
          },
          Object {
            "id": "2019",
            "label": "2019",
          },
          Object {
            "id": "2020",
            "label": "2020",
          },
        ],
        "filterName": "launchYear",
        "sectionName": "Launch Year",
      }
    }
  />,
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "true",
            "label": "True",
          },
          Object {
            "id": "false",
            "label": "False",
          },
        ],
        "filterName": "sucessfulLaunch",
        "sectionName": "Successful Launch",
      }
    }
  />,
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "true",
            "label": "True",
          },
          Object {
            "id": "false",
            "label": "False",
          },
        ],
        "filterName": "sucessfulLanding",
        "sectionName": "Successful Landing",
      }
    }
  />,
]
`;

exports[`Test the Filter Component 1`] = `
Array [
  "Filters",
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "2006",
            "label": "2006",
          },
          Object {
            "id": "2007",
            "label": "2007",
          },
          Object {
            "id": "2008",
            "label": "2008",
          },
          Object {
            "id": "2009",
            "label": "2009",
          },
          Object {
            "id": "2010",
            "label": "2010",
          },
          Object {
            "id": "2011",
            "label": "2011",
          },
          Object {
            "id": "2012",
            "label": "2012",
          },
          Object {
            "id": "2013",
            "label": "2013",
          },
          Object {
            "id": "2014",
            "label": "2014",
          },
          Object {
            "id": "2015",
            "label": "2015",
          },
          Object {
            "id": "2016",
            "label": "2016",
          },
          Object {
            "id": "2017",
            "label": "2017",
          },
          Object {
            "id": "2018",
            "label": "2018",
          },
          Object {
            "id": "2019",
            "label": "2019",
          },
          Object {
            "id": "2020",
            "label": "2020",
          },
        ],
        "filterName": "launchYear",
        "sectionName": "Launch Year",
      }
    }
  />,
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "true",
            "label": "True",
          },
          Object {
            "id": "false",
            "label": "False",
          },
        ],
        "filterName": "sucessfulLaunch",
        "sectionName": "Successful Launch",
      }
    }
  />,
  <FilterSection
    sectionConfig={
      Object {
        "buttons": Array [
          Object {
            "id": "true",
            "label": "True",
          },
          Object {
            "id": "false",
            "label": "False",
          },
        ],
        "filterName": "sucessfulLanding",
        "sectionName": "Successful Landing",
      }
    }
  />,
]
`;
